Fix file collections and file trees with relative files under CC

alllex opened the pull request:
Fixes https://github.com/gradle/gradle/issues/30052

### Problem

The root cause of the problem is that CC does not store the base directory for some file collection-related types. This results in it behaving differently from vintage: when a relative file needs to be resolved _after_ CC load happened, it can be potentially resolved against the wrong base directory.

Specifically, due to how CC injects the file collection factory service, the base directory was always the root-directory-of-the-build. This meant that only files relative to other base directories (like in subprojects) would be incorrectly resolved.

### Solution

This PR introduces a CC codec for the `PathToFileResolver` to be able to store the resolvers associated with (a) `ConfigurableFileCollection` and `ConfigurableFileTree`, as well as for (b) `ProviderBackedFileCollection`.

The (a) case requires the change, because these mutable containers of files can still be modified at execution time (after CC deserialization), and the relative files can still be added. The relative files must be resolved using the resolver equivalent to the one used to create the collection in the first place. It's similar for the file tree, where only the root can be changed to another relative file.

The (b) case is similar, but more likely to happen in practice. The provider-backed collection can yield relative paths, e.g. as plain strings, and those are currently evaluated after serialization (CC encodes the entire provider, without evaluating it). As such, those relative paths must also be resolved against an appropriate base directory, which then has to be encoded.

This change aligns the behavior between CC and vintage.

alllex commented:
@bot-gradle test without pts

alllex commented:
@bot-gradle test APT

bot-gradle commented:
<!-- {"replyTargetCommentId":2660317131,"triggeredTeamCityBuilds":[]} -->
Sorry some internal error occurs, please contact the administrator @gradle/bt-developer-productivity

alllex commented:
@bot-gradle test APT

bot-gradle commented:
<!-- {"replyTargetCommentId":2660318364,"triggeredTeamCityBuilds":[]} -->
Sorry some internal error occurs, please contact the administrator @gradle/bt-developer-productivity

bot-gradle commented:
<!-- {"replyTargetCommentId":2660319920,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_AllPerformanceTests_Trigger","buildId":"94860354","headRef":"2f2f287f66e2ef1622b1cced5911a1c94c281fb2"}]} -->
I've triggered the following builds:
- [All Performance Tests (Trigger)](https://builds.gradle.org/build/94860354)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=2f2f287f66e2ef1622b1cced5911a1c94c281fb2,ReadyforRelease&amp;)

bot-gradle commented:
The following builds have passed:
- [All Performance Tests (Trigger)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_AllPerformanceTests_Trigger/94860354)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=2f2f287f66e2ef1622b1cced5911a1c94c281fb2,ReadyforRelease&amp;)

alllex commented:
@bot-gradle test without pts

bot-gradle commented:
<!-- {"replyTargetCommentId":2679226918,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_PullRequestFeedback_Trigger","buildId":"95194787","headRef":"102f04061044489fb7ac7fe3001821a9d2b56cae"}]} -->
I've triggered the following builds:
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/build/95194787)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=102f04061044489fb7ac7fe3001821a9d2b56cae,PullRequestFeedback&amp;)

bot-gradle commented:
The following builds have failed:
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_PullRequestFeedback_Trigger/95194787)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=102f04061044489fb7ac7fe3001821a9d2b56cae&amp;)

alllex commented:
@bot-gradle test without pts

bot-gradle commented:
<!-- {"replyTargetCommentId":2679612355,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_PullRequestFeedback_Trigger","buildId":"95198848","headRef":"6b71b2f1cda6d7a87fb6b96dcf9f376ea0e7b285"}]} -->
I've triggered the following builds:
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/build/95198848)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=6b71b2f1cda6d7a87fb6b96dcf9f376ea0e7b285,PullRequestFeedback&amp;)

bot-gradle commented:
The following builds have failed:
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_PullRequestFeedback_Trigger/95198848)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=6b71b2f1cda6d7a87fb6b96dcf9f376ea0e7b285&amp;)

6hundreds commented on platforms/core-configuration/file-collections/src/integTest/groovy/org/gradle/api/file/FileTreeRelativePathIntegrationTest.groovy:
```diff
@@ -0,0 +1,60 @@
+    // Tests a use case that will be forbidden by IP
+    @Requires(IntegTestPreconditions.NotIsolatedProjects)
+    @Issue("https://github.com/gradle/gradle/issues/30052")
+    def "ConfigurableFileTree supports setting directory to relative path at execution time"() {
```
🤔 This test seems like a duplication of `ConfigurableFileCollection`'s test with same scenario. Doesn't it make sense to merge them in a data-driven test?

alllex replied:
They are very similar, though different in a number of subtle ways. I presume that's the reason why most other integration tests in the package are split into different files for CFC and CFT.

I don't feel strongly. After trying out your suggestion, it does not look too bad. We can have a test class dedicated to relative file paths then.

I'll push the changes, following your suggestion

6hundreds commented:
I wondering how much bloated CC entries will be with this PR and how it impact CC hit performance than. Seems like we are adding not much - a boolean and a string, but given how many file collections may be stored in Android builds, it can be a thing. 

Although, I don't think that this will be a driving decision point, since this PR improves correctness of CC

6hundreds reviewed (approved):
LGTM with one question about the test

bamboo commented on platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/DefaultFilePropertyFactory.java:
```diff
@@ -444,5 +443,10 @@ public PathToFileResolver newResolver(File baseDir) {
+
+        @Override
+        public File getBaseDir() {
+            return directoryProvider.get().getAsFile();
```
❌ This looks incorrect. Wouldn't the base dir actually be `createResolver().getBaseDir()`?

In any case I wonder if it wouldn't be better to have the codec operate in terms of concrete types rather than on the (augmented) abstract interface...

alllex replied:
You are right, it misses the use of the parent resolver. This also means that the current test coverage is not enough. Though writing an integration test for this sort of requires knowing exactly how the implementation works. But it's better than nothing.

> In any case I wonder if it wouldn't be better to have the codec operate in terms of concrete types rather than on the (augmented) abstract interface...

Using a simple getter for the base dir looked like a minimal enough implementation, especially given that the abstract type is fully internal and that we didn't plan to reconstruct the same type via constructors, and instead use higher level services to do so.
However, coupled with the missing coverage for the case above, it does seem like it's not enough to store the base dir. We should serialize the directory provider as a provider, instead of making a decision to materialize it at serialization-time.

bamboo commented on platforms/core-configuration/core-serialization-codecs/src/main/kotlin/org/gradle/internal/serialize/codecs/core/PathToFileResolverCodec.kt:
```diff
@@ -0,0 +1,50 @@
+) : Codec<PathToFileResolver> {
+
+    override suspend fun WriteContext.encode(value: PathToFileResolver) {
+        val baseDir = value.baseDir
```
💭 Rather than augmenting the `PathToFileResolver` to expose the `baseDir`, have you considered type casing the concrete implementations like we for file collections, for example?

alllex replied:
I responded in another comment mostly. The simplistic approach didn't pan out, so I'll update the codec to rely on concrete implementations

alllex commented on platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/DefaultFilePropertyFactory.java:
```diff
@@ -413,36 +412,4 @@ public File transform(FileSystemLocation location) {
         }
     }
 
-    private static class DirectoryProviderPathToFileResolver implements PathToFileResolver {
```
ℹ️  Moved to it's own file

alllex commented:
I addressed the initial comments. PTAL!

I also discovered that there are more related problems: when files derived from `DirectoryProperty` or `RegularFileProperty` are added to file collections, the connection between them and the file collection is not preserved upon decoding. This results in file collections not seeing updated files, if those properties are mutated at execution time. It does not seem to be a very likely scenario, but it's a discrepancy with how vintage work. It's unlikely, because properties which are task inputs would be finalized before execution time, but it's possible to "sneak in" these properties via closure-capture in `doLast` blocks or as `@Internal` inputs of a task. Then they are not finalized.

I added a test for `DirectoryProperty` that captures vintage behavior, but it fails with CC today, so I disabled it for now. Possibly to be addressed in a follow up.

bamboo commented on platforms/core-configuration/core-serialization-codecs/src/main/kotlin/org/gradle/internal/serialize/codecs/core/ConfigurableFileTreeCodec.kt:
```diff
@@ -30,14 +32,17 @@ class ConfigurableFileTreeCodec(
     private val fileCollectionFactory: FileCollectionFactory
 ) : Codec<ConfigurableFileTree> {
     override suspend fun WriteContext.encode(value: ConfigurableFileTree) {
+        value as DefaultConfigurableFileTree
```
💅 `require(value is DefaultConfigurationFileTree)` is more expressive and the failure mode is clearer (a precondition violation instead of a class cast exception)

bamboo commented on platforms/core-configuration/core-serialization-codecs/src/main/kotlin/org/gradle/internal/serialize/codecs/core/ConfigurableFileTreeCodec.kt:
```diff
@@ -30,14 +32,17 @@ class ConfigurableFileTreeCodec(
     }
 
     override suspend fun ReadContext.decode(): ConfigurableFileTree {
+        val resolver = read() as PathToFileResolver
```
💅 `readNonNull<PathToFileResolver>()`

alllex replied:
Ah, interesting. I didn't discover this method. A couple of codecs I looked at didn't use it, and we have plenty of codecs that use `read() as ...` pattern.

bamboo commented on platforms/core-configuration/core-serialization-codecs/src/main/kotlin/org/gradle/internal/serialize/codecs/core/PathToFileResolverCodec.kt:
```diff
@@ -0,0 +1,74 @@
+                encodeBean(value)
+            }
+
+            else -> error("Unexpected type of ${PathToFileResolver::class.simpleName} for ${value.javaClass}")
```
👍

bamboo commented on platforms/core-configuration/core-serialization-codecs/src/main/kotlin/org/gradle/internal/serialize/codecs/core/PathToFileResolverCodec.kt:
```diff
@@ -0,0 +1,74 @@
+                writeByte(1)
+            }
+
+            is BaseDirFileResolver -> {
```
🚀 Would there be a performance benefit to `encodePreservingSharedIdentifyOf(value) { ... }` here?

alllex replied:
Good idea!
It should help when encoding, if there are many collections with the same resolver (say, for a given project). For decoding, it would also avoid duplication of those instances in memory.
Pushed the change

bamboo reviewed (approved):
👍

alllex commented:
@bot-gradle test on linux without pts

bot-gradle commented:
<!-- {"replyTargetCommentId":2694364581,"triggeredTeamCityBuilds":[{"buildConfigurationId":"Gradle_Master_Check_Stage_PullRequestFeedback_Linux_Trigger","buildId":"95363325","headRef":"fa0e3417532e782cc892fcc6cbbf180fb5780038"}]} -->
I've triggered the following builds:
- [Pull Request Feedback (Trigger)(Linux) (Check)](https://builds.gradle.org/build/95363325)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=fa0e3417532e782cc892fcc6cbbf180fb5780038,PullRequestFeedback&amp;)

alllex commented:
Strangely, this refactoring [commit](https://github.com/gradle/gradle/pull/32359/commits/60607d0c6181ec1ab954ba33dba2e9676df0864f) causes a non-trivial CC serialization error on load in the Android Santa tracker smoke test:
```
Error while reading task graph	
> Exception while loading configuration for :doodles-lib: Could not load the value of field `transformer` of `org.gradle.api.internal.provider.TransformBackedProvider` bean found in field `provider` of `org.gradle.api.internal.provider.MappingProvider` bean found in field `capturedArgs` of `java.lang.invoke.SerializedLambda` bean found in field `transformer` of `org.gradle.api.internal.provider.TransformBackedProvider` bean found in field `provider` of `org.gradle.api.internal.provider.MappingProvider` bean found in field `__lintApiDatabase__` of `com.android.build.gradle.internal.lint.EnvironmentVariableInputs_Decorated` bean found in field `__environmentVariableInputs__` of task `:doodles-lib:lintAnalyzeDebug` of type `com.android.build.gradle.internal.lint.AndroidLintAnalysisTask`.	
```

With the deepest exception cause being:
```
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 4
	at org.gradle.internal.serialize.codecs.core.jos.JavaObjectSerializationCodec.decode(JavaObjectSerializationCodec.kt:299)
	at org.gradle.internal.serialize.graph.codecs.BindingsBackedCodec.decode(BindingsBackedCodec.kt:80)
	at org.gradle.internal.serialize.graph.DefaultReadContext.read(Contexts.kt:279)
	at org.gradle.internal.serialize.graph.BeanPropertyExtensionsKt.readPropertyValue(BeanPropertyExtensions.kt:61)
	... 217 more
```

Build scan: https://ge.gradle.org/s/ahnpo5tfakp7y/tests/task/:smoke-test:configCacheSantaTrackerSmokeTest/details/org.gradle.smoketests.AndroidSantaTrackerLintSmokeTest/can%20lint%20Santa-Tracker%20(agp%3D8.10.0-dev)?top-execution=1

<details><summary>Full exception</summary>
<p>

```
Cause 2: org.gradle.internal.serialize.codecs.core.OperationException: Exception while loading configuration for :playgames: Could not load the value of field `transformer` of `org.gradle.api.internal.provider.TransformBackedProvider` bean found in field `provider` of `org.gradle.api.internal.provider.MappingProvider` bean found in field `capturedArgs` of `java.lang.invoke.SerializedLambda` bean found in field `transformer` of `org.gradle.api.internal.provider.TransformBackedProvider` bean found in field `provider` of `org.gradle.api.internal.provider.MappingProvider` bean found in field `__lintApiDatabase__` of `com.android.build.gradle.internal.lint.EnvironmentVariableInputs_Decorated` bean found in field `__environmentVariableInputs__` of task `:playgames:lintAnalyzeDebug` of type `com.android.build.gradle.internal.lint.AndroidLintAnalysisTask`.
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodecKt$asBuildOperation$1.run(WorkNodeCodec.kt:636)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$QueueWorker.execute(DefaultBuildOperationExecutor.java:161)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.runOperation(DefaultBuildOperationQueue.java:272)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.doRunBatch(DefaultBuildOperationQueue.java:253)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.lambda$runBatch$1(DefaultBuildOperationQueue.java:226)
	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:263)
	at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:127)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.runBatch(DefaultBuildOperationQueue.java:224)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.run(DefaultBuildOperationQueue.java:192)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.gradle.api.GradleException: Could not load the value of field `transformer` of `org.gradle.api.internal.provider.TransformBackedProvider` bean found in field `provider` of `org.gradle.api.internal.provider.MappingProvider` bean found in field `capturedArgs` of `java.lang.invoke.SerializedLambda` bean found in field `transformer` of `org.gradle.api.internal.provider.TransformBackedProvider` bean found in field `provider` of `org.gradle.api.internal.provider.MappingProvider` bean found in field `__lintApiDatabase__` of `com.android.build.gradle.internal.lint.EnvironmentVariableInputs_Decorated` bean found in field `__environmentVariableInputs__` of task `:playgames:lintAnalyzeDebug` of type `com.android.build.gradle.internal.lint.AndroidLintAnalysisTask`.
	at org.gradle.internal.serialize.graph.BeanPropertyExtensionsKt.readPropertyValue(BeanPropertyExtensions.kt:70)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader.readFieldOf(BeanPropertyReader.kt:93)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader.readFieldsOf(BeanPropertyReader.kt:73)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader.access$readFieldsOf(BeanPropertyReader.kt:35)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader$readFieldsOf$1.invokeSuspend(BeanPropertyReader.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlin.coroutines.SafeContinuation.resumeWith(SafeContinuationJvm.kt:41)
	at org.gradle.internal.serialize.graph.CombinatorsKt$reentrant$1$decodeLoop$$inlined$Continuation$1.resumeWith(Continuation.kt:78)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
	at kotlin.coroutines.ContinuationKt.startCoroutine(Continuation.kt:115)
	at org.gradle.internal.serialize.graph.CombinatorsKt$reentrant$1.decodeLoop(Combinators.kt:123)
	at org.gradle.internal.serialize.graph.CombinatorsKt$reentrant$1.decode(Combinators.kt:85)
	at org.gradle.internal.serialize.graph.codecs.BindingsBackedCodec.decode(BindingsBackedCodec.kt:80)
	at org.gradle.internal.serialize.graph.DefaultReadContext.read(Contexts.kt:279)
	at org.gradle.internal.serialize.graph.BeanPropertyExtensionsKt.readPropertyValue(BeanPropertyExtensions.kt:61)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader.readFieldOf(BeanPropertyReader.kt:93)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader.readFieldsOf(BeanPropertyReader.kt:73)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader.readStateOf(BeanPropertyReader.kt:66)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodec$readTask$2.invokeSuspend(TaskNodeCodec.kt:134)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodec$readTask$2.invoke(TaskNodeCodec.kt)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodec$readTask$2.invoke(TaskNodeCodec.kt)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodecKt.withTaskOf(TaskNodeCodec.kt:236)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodecKt.access$withTaskOf(TaskNodeCodec.kt:1)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodec.readTask(TaskNodeCodec.kt:128)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodec.decode(TaskNodeCodec.kt:79)
	at org.gradle.internal.serialize.graph.codecs.BindingsBackedCodec.decode(BindingsBackedCodec.kt:80)
	at org.gradle.internal.serialize.graph.DefaultReadContext.read(Contexts.kt:279)
	at org.gradle.internal.serialize.graph.CodecKt.readNonNull(Codec.kt:156)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.readNode(WorkNodeCodec.kt:405)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.readGroupedNodes(WorkNodeCodec.kt:347)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.access$readGroupedNodes(WorkNodeCodec.kt:81)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1$1.invokeSuspend(WorkNodeCodec.kt:252)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1$1.invoke(WorkNodeCodec.kt)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1$1.invoke(WorkNodeCodec.kt)
	at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invokeSuspend(Codec.kt:144)
	at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invoke(Codec.kt)
	at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invoke(Codec.kt)
	at org.gradle.internal.serialize.graph.RunningKt$runReadOperation$2.invokeSuspend(Running.kt:33)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlin.coroutines.ContinuationKt.startCoroutine(Continuation.kt:115)
	at org.gradle.internal.serialize.graph.RunningKt.runToCompletion(Running.kt:54)
	at org.gradle.internal.serialize.graph.RunningKt.runReadOperation(Running.kt:32)
	at org.gradle.internal.serialize.graph.CodecKt.readWith(Codec.kt:143)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1.invoke(WorkNodeCodec.kt:251)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1.invoke(WorkNodeCodec.kt:250)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodecKt$asBuildOperation$1.run(WorkNodeCodec.kt:634)
	... 17 more
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 4
	at org.gradle.internal.serialize.codecs.core.jos.JavaObjectSerializationCodec.decode(JavaObjectSerializationCodec.kt:299)
	at org.gradle.internal.serialize.graph.codecs.BindingsBackedCodec.decode(BindingsBackedCodec.kt:80)
	at org.gradle.internal.serialize.graph.DefaultReadContext.read(Contexts.kt:279)
	at org.gradle.internal.serialize.graph.BeanPropertyExtensionsKt.readPropertyValue(BeanPropertyExtensions.kt:61)
	... 62 more
Cause 3: org.gradle.internal.serialize.codecs.core.OperationException: Exception while loading configuration for :common: Could not load the value of field `transformer` of `org.gradle.api.internal.provider.TransformBackedProvider` bean found in field `provider` of `org.gradle.api.internal.provider.MappingProvider` bean found in field `capturedArgs` of `java.lang.invoke.SerializedLambda` bean found in field `transformer` of `org.gradle.api.internal.provider.TransformBackedProvider` bean found in field `provider` of `org.gradle.api.internal.provider.MappingProvider` bean found in field `__lintApiDatabase__` of `com.android.build.gradle.internal.lint.EnvironmentVariableInputs_Decorated` bean found in field `__environmentVariableInputs__` of task `:common:lintAnalyzeDebug` of type `com.android.build.gradle.internal.lint.AndroidLintAnalysisTask`.
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodecKt$asBuildOperation$1.run(WorkNodeCodec.kt:636)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$QueueWorker.execute(DefaultBuildOperationExecutor.java:161)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.runOperation(DefaultBuildOperationQueue.java:272)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.doRunBatch(DefaultBuildOperationQueue.java:253)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.lambda$runBatch$1(DefaultBuildOperationQueue.java:226)
	at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:125)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.runBatch(DefaultBuildOperationQueue.java:224)
	at org.gradle.internal.operations.DefaultBuildOperationQueue$WorkerRunnable.run(DefaultBuildOperationQueue.java:192)
	at org.gradle.internal.operations.DefaultBuildOperationQueue.waitForCompletion(DefaultBuildOperationQueue.java:105)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.executeInParallel(DefaultBuildOperationExecutor.java:106)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.runAllWithAccessToProjectState(DefaultBuildOperationExecutor.java:75)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.runAllWithAccessToProjectState(DefaultBuildOperationExecutor.java:70)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$runBuildOperations$1.invoke(WorkNodeCodec.kt:582)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$runBuildOperations$1.invoke(WorkNodeCodec.kt:581)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.handleBuildOperationExceptions(WorkNodeCodec.kt:281)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.runBuildOperations(WorkNodeCodec.kt:581)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.readNodes(WorkNodeCodec.kt:248)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.doRead(WorkNodeCodec.kt:131)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.readWork(WorkNodeCodec.kt:101)
	at org.gradle.internal.cc.impl.ConfigurationCacheState.readWorkGraph(ConfigurationCacheState.kt:521)
	at org.gradle.internal.cc.impl.ConfigurationCacheState.readBuildContent$configuration_cache(ConfigurationCacheState.kt:502)
	at org.gradle.internal.cc.impl.ConfigurationCacheState.readBuildState(ConfigurationCacheState.kt:353)
	at org.gradle.internal.cc.impl.ConfigurationCacheState.readBuildsInTree(ConfigurationCacheState.kt:315)
	at org.gradle.internal.cc.impl.ConfigurationCacheState.readRootBuild(ConfigurationCacheState.kt:288)
	at org.gradle.internal.cc.impl.ConfigurationCacheState.readRootBuildState(ConfigurationCacheState.kt:189)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readRootBuildStateFrom$1.invokeSuspend(DefaultConfigurationCacheIO.kt:227)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readRootBuildStateFrom$1.invoke(DefaultConfigurationCacheIO.kt)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readRootBuildStateFrom$1.invoke(DefaultConfigurationCacheIO.kt)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheStateWithSpecialDecoders$1.invokeSuspend(DefaultConfigurationCacheIO.kt:351)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheStateWithSpecialDecoders$1.invoke(DefaultConfigurationCacheIO.kt)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheStateWithSpecialDecoders$1.invoke(DefaultConfigurationCacheIO.kt)
	at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invokeSuspend(Codec.kt:144)
	at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invoke(Codec.kt)
	at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invoke(Codec.kt)
	at org.gradle.internal.serialize.graph.RunningKt$runReadOperation$2.invokeSuspend(Running.kt:33)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlin.coroutines.ContinuationKt.startCoroutine(Continuation.kt:115)
	at org.gradle.internal.serialize.graph.RunningKt.runToCompletion(Running.kt:54)
	at org.gradle.internal.serialize.graph.RunningKt.runReadOperation(Running.kt:32)
	at org.gradle.internal.serialize.graph.CodecKt.readWith(Codec.kt:143)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.withReadContextFor(DefaultConfigurationCacheIO.kt:503)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.withReadContextFor(DefaultConfigurationCacheIO.kt:495)
	at org.gradle.internal.cc.impl.ConfigurationCacheBuildTreeIO$DefaultImpls.withReadContextFor(ConfigurationCacheBuildTreeIO.kt:83)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.withReadContextFor(DefaultConfigurationCacheIO.kt:94)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.readConfigurationCacheStateWithSpecialDecoders(DefaultConfigurationCacheIO.kt:349)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.access$readConfigurationCacheStateWithSpecialDecoders(DefaultConfigurationCacheIO.kt:94)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheState$1$1.invoke(DefaultConfigurationCacheIO.kt:257)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheState$1$1.invoke(DefaultConfigurationCacheIO.kt:256)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.withSharedObjectDecoderFor(DefaultConfigurationCacheIO.kt:333)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.access$withSharedObjectDecoderFor(DefaultConfigurationCacheIO.kt:94)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheState$1.invoke(DefaultConfigurationCacheIO.kt:256)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO$readConfigurationCacheState$1.invoke(DefaultConfigurationCacheIO.kt:255)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.withStringDecoderFor(DefaultConfigurationCacheIO.kt:321)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.readConfigurationCacheState(DefaultConfigurationCacheIO.kt:255)
	at org.gradle.internal.cc.impl.DefaultConfigurationCacheIO.readRootBuildStateFrom(DefaultConfigurationCacheIO.kt:225)
	at org.gradle.internal.cc.impl.DefaultConfigurationCache$loadWorkGraph$1$1$storeLoadResult$1.invoke(DefaultConfigurationCache.kt:657)
	at org.gradle.internal.cc.impl.DefaultConfigurationCache$loadWorkGraph$1$1$storeLoadResult$1.invoke(DefaultConfigurationCache.kt:656)
	at org.gradle.internal.cc.impl.ConfigurationCacheRepository$StoreImpl$useForStateLoad$1.invoke(ConfigurationCacheRepository.kt:231)
	at org.gradle.internal.cc.impl.ConfigurationCacheRepository$StoreImpl$useForStateLoad$1.invoke(ConfigurationCacheRepository.kt:231)
	at org.gradle.internal.cc.impl.ConfigurationCacheRepository$StoreImpl$useForStateLoad$2.invoke(ConfigurationCacheRepository.kt:239)
	at org.gradle.internal.cc.impl.ConfigurationCacheRepository$StoreImpl$useForStateLoad$2.invoke(ConfigurationCacheRepository.kt:235)
	at org.gradle.internal.cc.impl.ConfigurationCacheRepository$withExclusiveAccessToCache$1.get(ConfigurationCacheRepository.kt:320)
	at org.gradle.cache.internal.LockOnDemandCrossProcessCacheAccess.withFileLock(LockOnDemandCrossProcessCacheAccess.java:90)
	at org.gradle.cache.internal.DefaultCacheCoordinator.withFileLock(DefaultCacheCoordinator.java:213)
	at org.gradle.cache.internal.DefaultPersistentDirectoryStore.withFileLock(DefaultPersistentDirectoryStore.java:147)
	at org.gradle.cache.internal.DefaultCacheFactory$ReferenceTrackingCache.withFileLock(DefaultCacheFactory.java:203)
	at org.gradle.internal.cc.impl.ConfigurationCacheRepository.withExclusiveAccessToCache(ConfigurationCacheRepository.kt:318)
	at org.gradle.internal.cc.impl.ConfigurationCacheRepository.access$withExclusiveAccessToCache(ConfigurationCacheRepository.kt:54)
	at org.gradle.internal.cc.impl.ConfigurationCacheRepository$StoreImpl.useForStateLoad(ConfigurationCacheRepository.kt:235)
	at org.gradle.internal.cc.impl.ConfigurationCacheRepository$StoreImpl.useForStateLoad(ConfigurationCacheRepository.kt:231)
	at org.gradle.internal.cc.impl.DefaultConfigurationCache$loadWorkGraph$1$1.invoke(DefaultConfigurationCache.kt:656)
	at org.gradle.internal.cc.impl.DefaultConfigurationCache$loadWorkGraph$1$1.invoke(DefaultConfigurationCache.kt:655)
	at org.gradle.configurationcache.ConfigurationCacheBuildOperationsKt$withWorkGraphLoadOperation$1.call(ConfigurationCacheBuildOperations.kt:47)
	at org.gradle.configurationcache.ConfigurationCacheBuildOperationsKt$withWorkGraphLoadOperation$1.call(ConfigurationCacheBuildOperations.kt:40)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:210)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:205)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:54)
	at org.gradle.configurationcache.ConfigurationCacheBuildOperationsKt.withWorkGraphLoadOperation(ConfigurationCacheBuildOperations.kt:40)
	at org.gradle.internal.cc.impl.DefaultConfigurationCache.loadWorkGraph(DefaultConfigurationCache.kt:655)
	at org.gradle.internal.cc.impl.DefaultConfigurationCache.loadRequestedTasks(DefaultConfigurationCache.kt:253)
	at org.gradle.internal.cc.impl.ConfigurationCacheAwareBuildTreeWorkController$scheduleAndRunRequestedTasks$3.apply(ConfigurationCacheAwareBuildTreeWorkController.kt:69)
	at org.gradle.internal.cc.impl.ConfigurationCacheAwareBuildTreeWorkController$scheduleAndRunRequestedTasks$3.apply(ConfigurationCacheAwareBuildTreeWorkController.kt:68)
	at org.gradle.composite.internal.DefaultIncludedBuildTaskGraph.withNewWorkGraph(DefaultIncludedBuildTaskGraph.java:112)
	at org.gradle.internal.cc.impl.ConfigurationCacheAwareBuildTreeWorkController.scheduleAndRunRequestedTasks(ConfigurationCacheAwareBuildTreeWorkController.kt:68)
	at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.lambda$scheduleAndRunTasks$1(DefaultBuildTreeLifecycleController.java:77)
	at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.lambda$runBuild$4(DefaultBuildTreeLifecycleController.java:120)
	at org.gradle.internal.model.StateTransitionController.lambda$transition$6(StateTransitionController.java:169)
	at org.gradle.internal.model.StateTransitionController.doTransition(StateTransitionController.java:266)
	at org.gradle.internal.model.StateTransitionController.lambda$transition$7(StateTransitionController.java:169)
	at org.gradle.internal.work.DefaultSynchronizer.withLock(DefaultSynchronizer.java:46)
	at org.gradle.internal.model.StateTransitionController.transition(StateTransitionController.java:169)
	at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.runBuild(DefaultBuildTreeLifecycleController.java:117)
	at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.scheduleAndRunTasks(DefaultBuildTreeLifecycleController.java:77)
	at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.scheduleAndRunTasks(DefaultBuildTreeLifecycleController.java:72)
	at org.gradle.tooling.internal.provider.runner.BuildModelActionRunner.run(BuildModelActionRunner.java:53)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.internal.buildtree.ProblemReportingBuildActionRunner.run(ProblemReportingBuildActionRunner.java:49)
	at org.gradle.launcher.exec.BuildOutcomeReportingBuildActionRunner.run(BuildOutcomeReportingBuildActionRunner.java:71)
	at org.gradle.tooling.internal.provider.FileSystemWatchingBuildActionRunner.run(FileSystemWatchingBuildActionRunner.java:135)
	at org.gradle.launcher.exec.BuildCompletionNotifyingBuildActionRunner.run(BuildCompletionNotifyingBuildActionRunner.java:41)
	at org.gradle.launcher.exec.RootBuildLifecycleBuildActionExecutor.lambda$execute$0(RootBuildLifecycleBuildActionExecutor.java:54)
	at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:130)
	at org.gradle.launcher.exec.RootBuildLifecycleBuildActionExecutor.execute(RootBuildLifecycleBuildActionExecutor.java:54)
	at org.gradle.internal.buildtree.InitDeprecationLoggingActionExecutor.execute(InitDeprecationLoggingActionExecutor.java:62)
	at org.gradle.internal.buildtree.InitProblems.execute(InitProblems.java:36)
	at org.gradle.internal.buildtree.DefaultBuildTreeContext.execute(DefaultBuildTreeContext.java:40)
	at org.gradle.launcher.exec.BuildTreeLifecycleBuildActionExecutor.lambda$execute$0(BuildTreeLifecycleBuildActionExecutor.java:71)
	at org.gradle.internal.buildtree.BuildTreeState.run(BuildTreeState.java:60)
	at org.gradle.launcher.exec.BuildTreeLifecycleBuildActionExecutor.execute(BuildTreeLifecycleBuildActionExecutor.java:71)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor$2.call(RunAsBuildOperationBuildActionExecutor.java:67)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor$2.call(RunAsBuildOperationBuildActionExecutor.java:63)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:210)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:205)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:54)
	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor.execute(RunAsBuildOperationBuildActionExecutor.java:63)
	at org.gradle.launcher.exec.RunAsWorkerThreadBuildActionExecutor.lambda$execute$0(RunAsWorkerThreadBuildActionExecutor.java:36)
	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:263)
	at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:127)
	at org.gradle.launcher.exec.RunAsWorkerThreadBuildActionExecutor.execute(RunAsWorkerThreadBuildActionExecutor.java:36)
	at org.gradle.tooling.internal.provider.continuous.ContinuousBuildActionExecutor.execute(ContinuousBuildActionExecutor.java:110)
	at org.gradle.tooling.internal.provider.SubscribableBuildActionExecutor.execute(SubscribableBuildActionExecutor.java:64)
	at org.gradle.internal.session.DefaultBuildSessionContext.execute(DefaultBuildSessionContext.java:46)
	at org.gradle.internal.buildprocess.execution.BuildSessionLifecycleBuildActionExecutor$ActionImpl.apply(BuildSessionLifecycleBuildActionExecutor.java:92)
	at org.gradle.internal.buildprocess.execution.BuildSessionLifecycleBuildActionExecutor$ActionImpl.apply(BuildSessionLifecycleBuildActionExecutor.java:80)
	at org.gradle.internal.session.BuildSessionState.run(BuildSessionState.java:73)
	at org.gradle.internal.buildprocess.execution.BuildSessionLifecycleBuildActionExecutor.execute(BuildSessionLifecycleBuildActionExecutor.java:62)
	at org.gradle.internal.buildprocess.execution.BuildSessionLifecycleBuildActionExecutor.execute(BuildSessionLifecycleBuildActionExecutor.java:41)
	at org.gradle.internal.buildprocess.execution.StartParamsValidatingActionExecutor.execute(StartParamsValidatingActionExecutor.java:64)
	at org.gradle.internal.buildprocess.execution.StartParamsValidatingActionExecutor.execute(StartParamsValidatingActionExecutor.java:32)
	at org.gradle.internal.buildprocess.execution.SessionFailureReportingActionExecutor.execute(SessionFailureReportingActionExecutor.java:51)
	at org.gradle.internal.buildprocess.execution.SessionFailureReportingActionExecutor.execute(SessionFailureReportingActionExecutor.java:39)
	at org.gradle.internal.buildprocess.execution.SetupLoggingActionExecutor.execute(SetupLoggingActionExecutor.java:47)
	at org.gradle.internal.buildprocess.execution.SetupLoggingActionExecutor.execute(SetupLoggingActionExecutor.java:31)
	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:70)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:39)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:29)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:35)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.lambda$execute$0(ForwardClientInput.java:40)
	at org.gradle.internal.daemon.clientinput.ClientInputForwarder.forwardInput(ClientInputForwarder.java:80)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:64)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:63)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:84)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:52)
	at org.gradle.launcher.daemon.server.DaemonStateCoordinator.lambda$runCommand$0(DaemonStateCoordinator.java:321)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.gradle.api.GradleException: Could not load the value of field `transformer` of `org.gradle.api.internal.provider.TransformBackedProvider` bean found in field `provider` of `org.gradle.api.internal.provider.MappingProvider` bean found in field `capturedArgs` of `java.lang.invoke.SerializedLambda` bean found in field `transformer` of `org.gradle.api.internal.provider.TransformBackedProvider` bean found in field `provider` of `org.gradle.api.internal.provider.MappingProvider` bean found in field `__lintApiDatabase__` of `com.android.build.gradle.internal.lint.EnvironmentVariableInputs_Decorated` bean found in field `__environmentVariableInputs__` of task `:common:lintAnalyzeDebug` of type `com.android.build.gradle.internal.lint.AndroidLintAnalysisTask`.
	at org.gradle.internal.serialize.graph.BeanPropertyExtensionsKt.readPropertyValue(BeanPropertyExtensions.kt:70)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader.readFieldOf(BeanPropertyReader.kt:93)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader.readFieldsOf(BeanPropertyReader.kt:73)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader.access$readFieldsOf(BeanPropertyReader.kt:35)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader$readFieldsOf$1.invokeSuspend(BeanPropertyReader.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlin.coroutines.SafeContinuation.resumeWith(SafeContinuationJvm.kt:41)
	at org.gradle.internal.serialize.graph.CombinatorsKt$reentrant$1$decodeLoop$$inlined$Continuation$1.resumeWith(Continuation.kt:78)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
	at kotlin.coroutines.ContinuationKt.startCoroutine(Continuation.kt:115)
	at org.gradle.internal.serialize.graph.CombinatorsKt$reentrant$1.decodeLoop(Combinators.kt:123)
	at org.gradle.internal.serialize.graph.CombinatorsKt$reentrant$1.decode(Combinators.kt:85)
	at org.gradle.internal.serialize.graph.codecs.BindingsBackedCodec.decode(BindingsBackedCodec.kt:80)
	at org.gradle.internal.serialize.graph.DefaultReadContext.read(Contexts.kt:279)
	at org.gradle.internal.serialize.graph.BeanPropertyExtensionsKt.readPropertyValue(BeanPropertyExtensions.kt:61)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader.readFieldOf(BeanPropertyReader.kt:93)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader.readFieldsOf(BeanPropertyReader.kt:73)
	at org.gradle.internal.serialize.beans.services.BeanPropertyReader.readStateOf(BeanPropertyReader.kt:66)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodec$readTask$2.invokeSuspend(TaskNodeCodec.kt:134)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodec$readTask$2.invoke(TaskNodeCodec.kt)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodec$readTask$2.invoke(TaskNodeCodec.kt)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodecKt.withTaskOf(TaskNodeCodec.kt:236)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodecKt.access$withTaskOf(TaskNodeCodec.kt:1)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodec.readTask(TaskNodeCodec.kt:128)
	at org.gradle.internal.serialize.codecs.core.TaskNodeCodec.decode(TaskNodeCodec.kt:79)
	at org.gradle.internal.serialize.graph.codecs.BindingsBackedCodec.decode(BindingsBackedCodec.kt:80)
	at org.gradle.internal.serialize.graph.DefaultReadContext.read(Contexts.kt:279)
	at org.gradle.internal.serialize.graph.CodecKt.readNonNull(Codec.kt:156)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.readNode(WorkNodeCodec.kt:405)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.readGroupedNodes(WorkNodeCodec.kt:347)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec.access$readGroupedNodes(WorkNodeCodec.kt:81)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1$1.invokeSuspend(WorkNodeCodec.kt:252)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1$1.invoke(WorkNodeCodec.kt)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1$1.invoke(WorkNodeCodec.kt)
	at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invokeSuspend(Codec.kt:144)
	at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invoke(Codec.kt)
	at org.gradle.internal.serialize.graph.CodecKt$readWith$1$1.invoke(Codec.kt)
	at org.gradle.internal.serialize.graph.RunningKt$runReadOperation$2.invokeSuspend(Running.kt:33)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlin.coroutines.ContinuationKt.startCoroutine(Continuation.kt:115)
	at org.gradle.internal.serialize.graph.RunningKt.runToCompletion(Running.kt:54)
	at org.gradle.internal.serialize.graph.RunningKt.runReadOperation(Running.kt:32)
	at org.gradle.internal.serialize.graph.CodecKt.readWith(Codec.kt:143)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1.invoke(WorkNodeCodec.kt:251)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodec$readNodes$1$1$1.invoke(WorkNodeCodec.kt:250)
	at org.gradle.internal.serialize.codecs.core.WorkNodeCodecKt$asBuildOperation$1.run(WorkNodeCodec.kt:634)
	... 172 more
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 4
	at org.gradle.internal.serialize.codecs.core.jos.JavaObjectSerializationCodec.decode(JavaObjectSerializationCodec.kt:299)
	at org.gradle.internal.serialize.graph.codecs.BindingsBackedCodec.decode(BindingsBackedCodec.kt:80)
	at org.gradle.internal.serialize.graph.DefaultReadContext.read(Contexts.kt:279)
	at org.gradle.internal.serialize.graph.BeanPropertyExtensionsKt.readPropertyValue(BeanPropertyExtensions.kt:61)
	... 217 more
```

</p>
</details> 

I'll revert that commit for now, since it's not required to address the original problem.
But it's something to investigate in the future.

bot-gradle commented:
The following builds have passed:
- [Pull Request Feedback (Trigger)(Linux) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_PullRequestFeedback_Linux_Trigger/95363325)
  - [Build Scan](https://ge.gradle.org/scans?search.tags=CI&amp;search.names=gitCommitId,PartOf&amp;search.values=fa0e3417532e782cc892fcc6cbbf180fb5780038,PullRequestFeedback&amp;)
